### PR TITLE
yuzu/game_list&multiplayer: Specify string conversions explicitly

### PR DESCRIPTION
**Pull request by FearlessTobi**
Also see https://github.com/yuzu-emu/yuzu/pull/2503.

Makes all top-level source files compilable with implicit Qt string conversions disabled. Now all that's left is to address main.cpp, which I'll do in a follow-up PR, as it's quite a large file, and would like to have those changes reviewed separately. The follow up change will also finally pass compilation flags that disable implicit Qt string conversions.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4992)

### CODE REVIEW

File: src/citra_qt/game_list.cpp [hunk 1]
```diff
@@ -579,25 +584,27 @@ void GameList::LoadCompatibilityList() {
         return;
     }
 
-    const QString string_content = content;
-    QJsonDocument json = QJsonDocument::fromJson(string_content.toUtf8());
-    QJsonArray arr = json.array();
+    const QJsonDocument json = QJsonDocument::fromJson(content);
+    const QJsonArray arr = json.array();
+
+    for (const QJsonValue value : arr) {
```
Comment (zhaowenlan1779) [inline src/citra_qt/game_list.cpp]: Why is this changed from `QJsonValueRef` to `QJsonValue`?

Comment (FearlessTobi) [reply src/citra_qt/game_list.cpp]: @lioncash Could you answer this please?
